clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java

```diff
@@ -1122,7 +1122,7 @@ public void commitSync() {
     public void commitSync(final Map<TopicPartition, OffsetAndMetadata> offsets) {
         acquire();
         try {
-            coordinator.commitOffsetsSync(offsets, Long.MAX_VALUE);
+            coordinator.commitOffsetsSync(new HashMap<>(offsets), Long.MAX_VALUE);
```

Comment:
It's a bit annoying that this means a second copy when using `KafkaConsumer.commitSync` with no arguments. Maybe we could update that method so that it invokes `coordinator.commitOffsetsSync` directly instead of going through this method?